src/presentation/api/controller/ssl.go

```diff
@@ -0,0 +1,185 @@
+)
+
+type SslController struct {
+	persistentDbSvc *internalDbInfra.PersistentDatabaseService
+	transientDbSvc  *internalDbInfra.TransientDatabaseService
+	sslService      *service.SslService
+}
+
+func NewSslController(
+	persistentDbSvc *internalDbInfra.PersistentDatabaseService,
+	transientDbSvc *internalDbInfra.TransientDatabaseService,
+) *SslController {
+	return &SslController{
+		persistentDbSvc: persistentDbSvc,
+		transientDbSvc:  transientDbSvc,
+		sslService:      service.NewSslService(persistentDbSvc, transientDbSvc),
+	}
+}
+
+// ReadSslPairs	 godoc
+// @Summary      ReadSslPairs
+// @Description  List ssl pairs.
+// @Tags         ssl
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Success      200 {array} entity.SslPair
+// @Router       /v1/ssl/ [get]
+func (controller *SslController) Read(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(c, controller.sslService.Read())
+}
+
+func parseRawVhosts(rawVhostsInput interface{}) (rawVhosts []string, err error) {
+	rawVhostsSlice, assertOk := rawVhostsInput.([]interface{})
+	if !assertOk {
+		rawVhostUniqueStr, err := voHelper.InterfaceToString(rawVhostsInput)
+		if err != nil {
+			return rawVhosts, errors.New("VirtualHostsMustBeStringOrStringSlice")
+		}
+		return append(rawVhosts, rawVhostUniqueStr), err
+	}
+
+	rawVhosts = []string{}
+	for _, rawVhost := range rawVhostsSlice {
+		rawVhostStr, err := voHelper.InterfaceToString(rawVhost)
+		if err != nil {
+			slog.Debug(err.Error(), slog.Any("vhost", rawVhost))
+			continue
+		}
+		rawVhosts = append(rawVhosts, rawVhostStr)
+	}
+
+	return rawVhosts, err
+}
+
+// CreateSslPair    	 godoc
+// @Summary      CreateSslPair
+// @Description  Create a new ssl pair.
+// @Tags         ssl
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        createSslPairDto 	  body    dto.CreateSslPair  true  "All props are required.<br />virtualHosts may be string or []string. Alias is not allowed.<br />certificate is a string field, i.e. ignore the structure shown.<br />certificate and key must be base64 encoded."
+// @Success      201 {object} object{} "SslPairCreated"
+// @Router       /v1/ssl/ [post]
+func (controller *SslController) Create(c echo.Context) error {
+	requestBody, err := apiHelper.ReadRequestBody(c)
+	if err != nil {
+		return err
+	}
+
+	rawVhosts, err := parseRawVhosts(requestBody["virtualHosts"])
+	if err != nil {
+		return apiHelper.ResponseWrapper(c, http.StatusBadRequest, err.Error())
+	}
+	requestBody["virtualHosts"] = rawVhosts
+
+	encodedCert, err := valueObject.NewEncodedContent(requestBody["certificate"])
+	if err != nil {
+		return apiHelper.ResponseWrapper(c, http.StatusBadRequest, err.Error())
+	}
+	decodedCert, err := encodedCert.GetDecodedContent()
+	if err != nil {
+		return apiHelper.ResponseWrapper(
+			c, http.StatusBadRequest, "CannotDecodeSslCertificateContent",
+		)
+	}
+	requestBody["certificate"] = decodedCert
```

Comment:
Why do you encode and then decode? If you want to pass the cert content to the service as string, you may pass what the user sent you directly and let the service make sure the content is valid.

Reply:
I think there was a bit of confusion due to the **VO**'s name. In fact, I receive the encoded value and pass it into the **VO** constructor so it can validate whether the value is properly encoded. If it is, I then decode it and send it to my service.

I handle this in the **API** because its input differs from the **CLI**, which receives the path of a file and reads its content as unencoded. Therefore, I can't send the encoded value directly to the service, as it wouldn't be able to process the certificate correctly.